diff --git a/uuid.entity.inc b/uuid.entity.inc
index 4353cc6..8efa786 100644
--- a/uuid.entity.inc
+++ b/uuid.entity.inc
@@ -88,17 +88,17 @@ function uuid_entity_info_alter(&$info) {
  * This adds the UUID as an entity property for all UUID-enabled entities
  * which automatically gives us token and Rules integration.
  */
-function uuid_entity_property_info_alter(&$info) {
+function uuid_entity_property_info_alter(&$data) {
   foreach (entity_get_info() as $entity_type => $info) {
     if (isset($info['uuid']) && $info['uuid'] == TRUE && !empty($info['entity keys']['uuid'])) {
-      $info[$entity_type]['properties'][$info['entity keys']['uuid']] = array(
+      $data[$entity_type]['properties'][$info['entity keys']['uuid']] = array(
         'label' => t('UUID'),
         'type' => 'text',
         'description' => t('The universally unique ID.'),
         'schema field' => $info['entity keys']['uuid'],
       );
       if (!empty($info['entity keys']['revision uuid'])) {
-        $info[$entity_type]['properties'][$info['entity keys']['revision uuid']] = array(
+        $data[$entity_type]['properties'][$info['entity keys']['revision uuid']] = array(
           'label' => t('Revision UUID'),
           'type' => 'text',
           'description' => t("The revision's universally unique ID."),
