diff --git a/simpletest_clone_test_case.php b/simpletest_clone_test_case.php
index 67226ac..da265f2 100644
--- a/simpletest_clone_test_case.php
+++ b/simpletest_clone_test_case.php
@@ -105,6 +105,11 @@ class SimpleTestCloneTestCase extends DrupalWebTestCase {
       $this->cloneTable($name, $sources[$name], $schema);
     }
 
+    // Set path variables.
+    variable_set('file_public_path', $public_files_directory);
+    variable_set('file_private_path', $private_files_directory);
+    variable_set('file_temporary_path', $temp_files_directory);
+
     // Log in with a clean $user.
     $this->originalUser = $user;
     drupal_save_session(FALSE);
