diff --git a/simpletest_clone_test_case.php b/simpletest_clone_test_case.php
index 640353a..cb88f9b 100644
--- a/simpletest_clone_test_case.php
+++ b/simpletest_clone_test_case.php
@@ -118,6 +118,7 @@ class SimpleTestCloneTestCase extends DrupalWebTestCase {
     variable_set('mail_system', array('default-system' => 'TestingMailSystem'));

     drupal_set_time_limit($this->timeLimit);
+    $this->setup = TRUE;
   }

   /**
