diff --git a/simpletest_clone_test_case.php b/simpletest_clone_test_case.php
index da265f2..b300bbd 100644
--- a/simpletest_clone_test_case.php
+++ b/simpletest_clone_test_case.php
@@ -123,6 +123,8 @@ class SimpleTestCloneTestCase extends DrupalWebTestCase {
     variable_set('mail_system', array('default-system' => 'TestingMailSystem'));
 
     drupal_set_time_limit($this->timeLimit);
+    $this->resetAll();
+    $this->refreshVariables();
     $this->setup = TRUE;
   }
 
