diff --git a/modules/user/user.module b/modules/user/user.module
index 622fe4d..499dccb 100644
--- a/modules/user/user.module
+++ b/modules/user/user.module
@@ -1700,9 +1700,11 @@ function user_menu() {

   $items['admin/people/create'] = array(
     'title' => 'Add user',
+    'page callback' => 'user_admin',
     'page arguments' => array('create'),
     'access arguments' => array('administer users'),
     'type' => MENU_LOCAL_ACTION,
+    'file' => 'user.admin.inc',
   );

   // Administration pages.
