diff --git a/features.export.inc b/features.export.inc
old mode 100644
new mode 100755
index 68c2da2..86b6abe
--- a/features.export.inc
+++ b/features.export.inc
@@ -304,7 +304,7 @@ function features_export_render($export, $module_name, $reset = FALSE) {
     $code['module'] = "<?php\n/**\n * @file\n * Code for the {$export['name']} feature.\n */\n\ninclude_once '{$module_name}.features.inc';\n";
   }
   else {
-    $code['module'] = "<?php\n/**\n * @file\n * Drupal needs this blank file.\n */\n";
+    $code['module'] = "<?php\n/**\n * @file\nDrupal needs this blank file.\n*/\n";
   }
   return $code;
 }
